src/Operator.ts

```diff
@@ -1,7 +1,11 @@
 import {Subscriber} from './Subscriber';
 
-export class Operator<T, R> {
-  call<T, R>(subscriber: Subscriber<R>): Subscriber<T> {
+export interface Operator<T, R> {
```

Comment:
do we expect export interface and implementation both under same name? would you able to elaborate use case for interface is required with same name of implementation?